### PR TITLE
✨[amp-consent] Implement `Ping` command for TCF PostMessage API

### PR DESCRIPTION
**Pull request by micajuine-ho**
Partial for #30385

Implement Ping command received by 3p iframes through the PostMessage API. iframes can be children of other iframes, and so we hold on to a reference of them to send the response to. In the future, with the other commands, we will need to store state of consent (addEventListener), and we will always need the PolicyManager.

We pass along the consent policy manager from amp consent, so that we can get the `gdprApplies` value (and other consent data in the future).

Returns a "minimal" Ping object via PostMessage to the original source:
```
{
 *  gdprApplies: (boolean|undefined),
// Always true, as the command will only be received after amp-consent loads. Assumed that if publisher is using the PostMessage API, then a CMP will be used as well.
 *  cmpLoaded: (boolean),
// Always 'loaded'
 *  cmpStatus: (string|undefined),
// Always 2
 *  tcfPolicyVersion: number,
}
```

Returns the Ping object in the defined [`__tcfapiReturn`](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#using-postmessage) object

### CODE REVIEW

File: extensions/amp-consent/0.1/tcf-api-commands.js [hunk 1]
```diff
@@ -22,19 +23,63 @@ const TAG = 'amp-consent';
 
 /**
  * @param {!Object} payload
+ * @param {!Window} source
```
Comment (dmanek) [inline extensions/amp-consent/0.1/tcf-api-commands.js:26]: nit: rename param to win/window?

File: extensions/amp-consent/0.1/tcf-api-commands.js [hunk 2]
```diff
@@ -22,19 +23,63 @@ const TAG = 'amp-consent';
...
+ * once object has been filled.
+ * @param {!Object} payload
+ * @param {!Window} source
+ * @param {!ConsentPolicyManager} policyManager
+ */
+export function handlePingEvent(payload, source, policyManager) {
+  const cmpLoaded = true;
+  const metadataPromise = policyManager.getConsentMetadataInfo('default');
+
+  metadataPromise.then((metadata) => {
```
Comment (dmanek) [inline extensions/amp-consent/0.1/tcf-api-commands.js:54]: Combine prev 2 lines `policyManager.getConsentMetadataInfo('default').then(...)`

File: extensions/amp-consent/0.1/tcf-api-commands.js (hunk 1 above)
Comment (micajuine-ho) [reply extensions/amp-consent/0.1/tcf-api-commands.js:26]: Ok

File: extensions/amp-consent/0.1/tcf-api-commands.js [hunk 3]
```diff
@@ -14,53 +14,147 @@
...
+    }
+    const {command, parameter, version} = payload;
+    if (!isEnumValue(TCF_POST_MESSAGE_API_COMMANDS, command)) {
+      user().error(
+        TAG,
+        `Unsupported command found in "tcfapiCall": ${command}`
+      );
+      return false;
+    }
+    if (parameter !== undefined) {
```
Comment (dmanek) [inline extensions/amp-consent/0.1/tcf-api-commands.js:138]: Is `null` an acceptable value for parameter? If no, then just change this to `if (parameter) {...}`

File: extensions/amp-consent/0.1/tcf-api-commands.js [hunk 4]
```diff
@@ -14,53 +14,147 @@
...
+      user().error(
+        TAG,
+        `Unsupported parameter found in "tcfapiCall": ${parameter}`
+      );
+      return false;
+    }
+    if (version != '2') {
```
Comment (dmanek) [inline extensions/amp-consent/0.1/tcf-api-commands.js:145]: `version !== '2'`

**Review by dmanek — APPROVED** [review]

**Pull request merged**